### ISSUE TITLE
"Hidden state" doesn't handle reverting the `visibility`

### ISSUE THREAD

**Issue by Jym77** [act-rules/act-rules.github.io#1656]
[Hidden state](https://act-rules.github.io/glossary/#hidden-state) of an element is `true` if any of its ancestor has `visibility: hidden`.

This means that in
```html
<div style="visibility: hidden">
  Hello <span style="visibility: visible"><span>World</span></span>
</div>
```
Both `span` have a "hidden state" of `true` even though they are actually visible (and included in the accessibility tree (well not really as `span` are only containers, but that's the idea)).
Since `visibility` is inherited, that condition should only look at the element, not on its ancestors.

Also, the def should handle `visibility: collapse`.

**Comment by WilcoFiers** [comment]
Good point! Any suggestions on how to resolve? Would you be up for opening a PR?

**Comment by Jym77** [comment]
Yes, the fix should be fairly easy-ish…

A bit unrelated, but I'm also not happy with the name "hidden state". "Hidden" is pretty much visibility vocabulary in vernacular language but the def is about not being exposed at all to AT.

Instead of "hidden state is true", we could say "the element is not exposed" or "the element is inert" or something in these lines…
